src/value.cc

```diff
@@ -113,17 +113,17 @@ Value::Value(double v) : value(v)
   //  std::cout << "creating double " << v << "\n";
 }
 
-Value::Value(const std::string &v) : value(v)
+Value::Value(const std::string &v) : value(str_utf8_wrapper(v))
```

Comment:
This will initialize the `value` member, which is a boost::variant. This will most likely not make a copy. I would assume that the boost::variant constructor is capable of using move semantics and only construct one wrapper object.

This may actually be a good reason to use uniform initialization instead, as well as pass the string by value:
`Value::Value(std::string v) : value{str_utf8_wrapper{std::move(v)}}`

For a good explanation, see Page 281, in Effective Modern C++, "Item 41: Consider pass by value for copyable parameters that are cheap to move and always copied": https://edisciplinas.usp.br/pluginfile.php/1995323/mod_resource/content/1/Effective%20Modern%20C%2B%2B%202014.pdf